### ISSUE TITLE
trim-newlines ReDoS security issue in @wdio/devtools-service

### ISSUE THREAD

**Issue by JordiSAGE** [webdriverio/webdriverio#7294]
trim-newlines ReDoS security issue in @wdio/devtools-service

Environment:
•	WebdriverIO version: 6.8.2
•	Mode: WDIO Testrunner (sync)
•	Node.js version: 12.19.0
•	NPM version: 6.14.8
•	Browser name and version: Chrome 91
•	Platform name and version: Windows 10
•	Additional wdio packages used (if applicable): @wdio/devtools-service 6.8.0

**Config of WebdriverIO**
exports.config = {
    runner: 'local',
    specs: 'test.feature',
    maxInstances: 1,
    capabilities: [
        {
            maxInstances: 1,
            browserName: 'chrome',
        },
    ],
    sync: true,
    deprecationWarnings: false,
    logLevel: 'warn',
    coloredLogs: false,
    screenshotPath: cfgScreenshotPath,
    bail: 0,
    baseUrl: cfgBaseUrl,
    waitforTimeout: cfgTimeout,
    connectionRetryTimeout: cfgTimeout,
    connectionRetryCount: cfgConnectionRetryCount,
    services: ['devtools', 'shared-store'],
    framework: 'cucumber',
    reporters: [
        'dot',
        'spec',
        [
            'allure',
            {
                outputDir: path.resolve(cfgReportDir + '/allure-results'),
                disableWebdriverStepsReporting: true,
                useCucumberStepReporter: true,
            },
        ],
    ],
    cucumberOpts: {
        require: cfgStepDefinitions,
        backtrace: true, 
        compiler: ['ts:ts-node/register', 'js:babel-register'],
        requireModule: [
            'tsconfig-paths/register',
            () => {
                require('ts-node').register({ files: true });
            },
        ],
        dryRun: false,
        failFast: cfgfailFast,
        format: ['pretty'],
        colors: true,
        snippets: true,
        source: true,
        profile: [],
        strict: true,
        tagsInTitle: false,
        timeout: cfgTimeout,
        ignoreUndefinedDefinitions: false,
        tagExpression: cucumberTags,
    },
    onPrepare: (config, capabilities) => {
        // Log details moved here to avoid duplication in every thread
        console.log('Features path: ' + featuresPath);
        console.log('Cucumber tags: ' + cucumberTags);
        console.log('Feature to run: ' + featureToRun);

        config.specs.forEach(featureFile => {
            console.log(featureFile);
        });
        rimraf.sync(path.resolve(cfgReportDir + '/allure-results'));
    },
    before: () => {
        require('bytenode');
        browser.setTimeout({ implicit: cfgTimeout });

        /**
         * Setup the Chai assertion framework
         */
        const chai = require('chai');

        global.expect = chai.expect;
        global.assert = chai.assert;
        global.should = chai.should();

        const puppeteer = browser.getPuppeteer();
        const page = browser.call(() => puppeteer.pages())[0];
        browser.call(() =>
            page._client.send('Page.setDownloadBehavior', {
                behavior: 'allow',
                downloadPath: process.env.downloadFolder,
            }),
        );
    },
    beforeSuite: () => {},
    beforeScenario: scenario => {},
    afterScenario: function (uri, feature, scenario, result, sourceLocation) {
        var featureName = uri.toString();
        var pathSeparator;
        if (featureName.includes('\\')) {
            pathSeparator = '\\';
        } else {
            pathSeparator = '/';
        }
        featureName = featureName.substring(featureName.lastIndexOf(pathSeparator) + 1);
        allureReporter.addArgument('Feature', featureName);
        if (result.status === 'failed') {
            browser.takeScreenshot();
        }
    },
    afterStep: function (uri, feature, { error, result, duration, passed }, stepData, context) {
        if (error !== undefined) {
            browser.takeScreenshot();
        }
        browser.pause(200);
    },
    afterFeature: suite => {},
    after: (result, capabilities, specs) => {
        createEnvironmentProperties();
    },
    onComplete: (exitCode, config) => {},
};

**Describe the bug:**
As part of our security checking process we detected that, both in wdio version 6 and 7 the “@wdio/devtools-service” is getting use of the Meow package (meow@3.7.0), that has in turn the trim-newlines@1.0.0 package as dependency, being this last package the one that produced the following security issue, detected by ThreadFix tool as High vulnerability:

•	Reference 1753
•	Component trim-newlines:1.0.0
•	Description `trim-newlines` before 3.0.1 and 4.x before 4.0.1 for Node.js has an issue related to regular expression denial-of-service (ReDoS) for the .end() method.
•	http://cve.mitre.org/cgi-bin/cvename.cgi?name=1753

trim-newlines@1.0.0 dependency affected:
+-- @wdio/devtools-service@7.9.1
| `-- speedline@1.4.3
|   `-- meow@3.7.0
|     `-- trim-newlines@1.0.0

**Due to our requeriments to be compliant with other applications we manage, it would be required to base the fix or to generate a new issue version based in webdriverio version 6 (We're currently using WebdriverIO version 6.8.2)**

**To Reproduce**
Check on ThreadFix security vulnerability platform or any other security tool.

**Expected behavior**
@wdio/devtools-service version 6 should include meow package above 3.7.0 and trim-newlines above 3.0.1 or 4.0.1

**Additional context**
We already contacted the "Security Steering Comitee" , three times, on July 6th, 15th and 28th without reply from them.
We understand that they could be busy with other matters, and we appoligize in case we're not addressing the request properly, but we're close to a new release date and this issue is stopping us to provide a new version, Thanks in advance for your understanding.

**Comment by christian-bromann** [comment]
Hey @JordiSAGE ,

thanks for filing the issue and apologies the TSC team didn't respond in time. We are all indeed very busy.

The dependency `meow` is used in Speedlines CLI which WebdriverIO is __not__ making use of. Therefor this vulnerability doesn't not affect any WebdriverIO user whatsoever. There is also no way to update the package given that `speedline` still depends on a very old version of `meow`.

I will close this given the vulnerability doesn't affect WebdriverIO users.
